mjit.h

```diff
@@ -0,0 +1,125 @@
+/**********************************************************************
+
+  mjit.h - Interface to MRI method JIT compiler
+
+  Copyright (C) 2017 Vladimir Makarov <vmakarov@redhat.com>.
+
+**********************************************************************/
+
+#ifndef RUBY_MJIT_H
+#define RUBY_MJIT_H 1
+
+#include "ruby.h"
+
+/* Special address values of a function generated from the
+   corresponding iseq by MJIT: */
+enum rb_mjit_iseq_func {
+    /* ISEQ was not queued yet for the machine code generation */
+    NOT_ADDED_JIT_ISEQ_FUNC = 0,
+    /* ISEQ is already queued for the machine code generation but the
+       code is not ready yet for the execution */
+    NOT_READY_JIT_ISEQ_FUNC = 1,
+    /* ISEQ included not compilable insn or some assertion failed  */
+    NOT_COMPILABLE_JIT_ISEQ_FUNC = 2,
+    /* End mark */
+    LAST_JIT_ISEQ_FUNC = 3,
+};
+
+/* C compiler used to generate native code. */
+enum rb_mjit_cc {
+    /* Not selected */
+    MJIT_CC_DEFAULT = 0,
+    /* GNU Compiler Collection */
+    MJIT_CC_GCC = 1,
+    /* LLVM/Clang */
+    MJIT_CC_CLANG = 2,
+};
+
+/* MJIT options which can be defined on the MRI command line.  */
+struct mjit_options {
+    char on; /* flag of MJIT usage  */
+    /* Default: clang for macOS, cl for Windows, gcc for others. */
+    enum rb_mjit_cc cc;
+    /* Save temporary files after MRI finish.  The temporary files
+       include the pre-compiled header, C code file generated for ISEQ,
+       and the corresponding object file.  */
+    char save_temps;
+    /* Print MJIT warnings to stderr.  */
+    char warnings;
+    /* Disable compiler optimization and add debug symbols. It can be
+       very slow.  */
+    char debug;
+    /* Force printing info about MJIT work of level VERBOSE or
+       less. 0=silence, 1=medium, 2=verbose.  */
+    int verbose;
+    /* Maximal permitted number of iseq JIT codes in a MJIT memory
+       cache.  */
+    int max_cache_size;
+};
+
+RUBY_SYMBOL_EXPORT_BEGIN
+extern struct mjit_options mjit_opts;
+extern int mjit_init_p;
+
+extern void mjit_add_iseq_to_process(const rb_iseq_t *iseq);
+RUBY_SYMBOL_EXPORT_END
+
+extern int mjit_compile(FILE *f, const struct rb_iseq_constant_body *body, const char *funcname);
+extern void mjit_init(struct mjit_options *opts);
+extern void mjit_finish();
```

Comment:
Some functions are declared/defined in old K&R style.

Reply:
Unified the style in https://github.com/ruby/ruby/pull/1782/commits/e22a8e263383b6f924b94d29f65b677344b5daa9